flink-core/src/main/java/org/apache/flink/util/IOUtils.java

```diff
@@ -216,9 +215,52 @@ public static void closeSocket(final Socket sock) {
 	}
 
 	/**
+	 * Closes all {@link AutoCloseable} objects in the parameter, suppressing exceptions. Exception will be emitted
+	 * after calling close() on every object.
+	 *
+	 * @param closeables iterable with closeables to close.
+	 * @throws Exception collected exceptions that occurred during closing
+	 */
+	public static void closeAll(Iterable<? extends AutoCloseable> closeables) throws Exception {
+		if (null != closeables) {
+
+			Exception collectedExceptions = null;
+
+			for (AutoCloseable closeable : closeables) {
+				try {
+					if (null != closeable) {
+						closeable.close();
+					}
+				} catch (Exception e) {
+					if (null == collectedExceptions) {
+						collectedExceptions = e;
+					} else {
+						collectedExceptions.addSuppressed(e);
+					}
```

Comment:
Here you could use `collectedExceptions = ExceptionUtils.firstOrSuppressed(e, collectedExceptions);`